### PR TITLE
Switch docs viewer to use trees as the selection mechanism. This allo…

### PR DESCRIPTION
**Pull request by BananaEarwig**
…ws us to restructure the docs folder hierarchially. Incorporate action handlers.

### CODE REVIEW

File: apps/docs/app.lisp [hunk 1]
```diff
@@ -22,46 +25,71 @@
...
+		(def *file_tree_scroll* :min_width w)
+		(.-> *file_tree_scroll* :layout :dirty_all)))
+
+;import actions
+(import "./actions.inc")
+
 (defun main ()
-	(populate-page (defq file (elem-get 0 doc_list)))
+	(defq select (alloc-select +select_size) *running* :t)
+	(. *file_tree* :populate "./docs" '(".md") 2)
```
Comment (vygr) [inline apps/docs/app.lisp:76]: use

`(. *file_tree* :populate "docs" '(".md"))`

File: apps/docs/app.lisp [hunk 2]
```diff
@@ -22,46 +25,71 @@
...
+	(populate-page "./docs/INTRO.md")
```
Comment (vygr) [inline apps/docs/app.lisp:77]: `(populate-page "docs/INTRO.md")`

File: apps/docs/ui.inc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+(defun action-file-leaf-action ()
+	;load up the file selected
+	(defq file (. *file_tree* :get_route
+		(. *window* :find_id (getf *msg* +ev_msg_action_source_id))))
+	(select-node file)
+	(populate-page file))
+
+(defun action-close ()
+	(setq *running* :nil))
+
```
Comment (vygr) [inline apps/docs/ui.inc:17]: does this not exist in edit/ui.inc ?

File: apps/docs/app.lisp (hunk 1 above)
Comment (BananaEarwig) [reply apps/docs/app.lisp:76]: will do

File: apps/docs/app.lisp (hunk 2 above)
Comment (BananaEarwig) [reply apps/docs/app.lisp:77]: will do

File: apps/docs/ui.inc (hunk 3 above)
Comment (BananaEarwig) [reply apps/docs/ui.inc:17]: It does but it relies on vdu which is not being used in the docs app.

Comment (vygr) [reply apps/docs/ui.inc:17]: ok

File: apps/docs/app.lisp [hunk 4]
```diff
@@ -22,46 +25,71 @@
...
+		(ui-flow _ (:flow_flags +flow_stack_fill)
+			(ui-scroll *file_tree_scroll* +scroll_flag_vertical :nil
+				(. (ui-tree *file_tree* +event_file_folder_action
+						(:min_width 0 :color +argb_white :font *env_medium_terminal_font*))
+					:connect +event_tree_action))
+			(ui-backdrop _ (:color +argb_white)))
 		(ui-scroll page_scroll +scroll_flag_vertical (:min_height 900))))
 
+(defun select-node (file)
+	(when :t
```
Comment (vygr) [inline apps/docs/app.lisp:59]: should that be `(when file)` or maybe not there at all ?

File: apps/docs/app.lisp [hunk 5]
```diff
@@ -22,46 +25,71 @@
...
+		(def *file_tree_scroll* :min_width w)
+		(.-> *file_tree_scroll* :layout :dirty_all)))
+
+;import actions
+(import "./actions.inc")
+
 (defun main ()
-	(populate-page (defq file (elem-get 0 doc_list)))
+	(defq select (alloc-select +select_size) *running* :t)
+	(. *file_tree* :populate "./docs" '(".md"))
```
Comment (vygr) [inline apps/docs/app.lisp:76]: "docs" not "./docs"

File: apps/docs/ui.inc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+(defun action-minimise ()
+	(bind '(x y w h) (apply view-fit
+		(cat (. *window* :get_pos) (. *window* :pref_size))))
+	(. *window* :change_dirty x y w h))
+
+(defun action-maximise ()
+	(bind '(x y) (. *window* :get_pos))
+	(bind '(w h) (. *window* :pref_size))
+	(bind '(x y w h) (view-fit x y (* w 2) (* h 2)))
```
Comment (vygr) [inline apps/docs/ui.inc:26]: I suspect that makeing the width `(* w 2)` won't have any effect toher than leave a nasty band down the right side of the page ?

Comment (BananaEarwig) [reply apps/docs/ui.inc:26]: strangely that doesn't happen but the minimize and maximize seem to be inverted. Investigating.

File: apps/docs/app.lisp (hunk 5 above)
Comment (BananaEarwig) [reply apps/docs/app.lisp:76]: done

File: apps/docs/ui.inc (hunk 6 above)
Comment (BananaEarwig) [reply apps/docs/ui.inc:26]: ok fixed min / max

File: apps/docs/app.lisp (hunk 4 above)
Comment (BananaEarwig) [reply apps/docs/app.lisp]: Done

**Pull request merged**